src/lib.rs

```diff
@@ -114,97 +114,3630 @@ use std::{
+        return Language::St(region);
+    } else if var_env.contains("tn") {
+        return Language::Tn(region);
+    } else if var_env.contains("sn") {
+        return Language::Sn(region);
+    } else if var_env.contains("sd") {
+        return Language::Sd(region);
+    } else if var_env.contains("si") {
+        return Language::Si(region);
+    } else if var_env.contains("ss") {
+        return Language::Ss(region);
+    } else if var_env.contains("sk") {
+        return Language::Sk(region);
+    } else if var_env.contains("sl") {
+        return Language::Sl(region);
+    } else if var_env.contains("so") {
+        return Language::So(region);
+    } else if var_env.contains("es") {
+        return Language::Es(region);
+    } else if var_env.contains("su") {
+        return Language::Su(region);
+    } else if var_env.contains("sw") {
+        return Language::Sw(region);
+    } else if var_env.contains("sv") {
+        return Language::Sv(region);
+    } else if var_env.contains("tl") {
+        return Language::Tl(region);
+    } else if var_env.contains("tg") {
+        return Language::Tg(region);
+    } else if var_env.contains("ta") {
+        return Language::Ta(region);
+    } else if var_env.contains("tt") {
+        return Language::Tt(region);
+    } else if var_env.contains("te") {
+        return Language::Te(region);
+    } else if var_env.contains("th") {
+        return Language::Th(region);
+    } else if var_env.contains("bo") {
+        return Language::Bo(region);
+    } else if var_env.contains("ti") {
+        return Language::Ti(region);
+    }
+
+    sixth_get_language_helper()
+}
+
+
+fn sixth_get_language_helper() -> Language {
+    let var_env: String = get_language_and_country();
+    let region: Region = Region::get();
+
+    if var_env.contains("to") {
+        return Language::To(region);
+    } else if var_env.contains("ts") {
+        return Language::Ts(region);
+    } else if var_env.contains("tr") {
+        return Language::Tr(region);
+    } else if var_env.contains("tk") {
+        return Language::Tk(region);
+    } else if var_env.contains("tw") {
+        return Language::Tw(region);
+    } else if var_env.contains("uk") {
+        return Language::Uk(region);
+    } else if var_env.contains("ur") {
+        return Language::Ur(region);
+    } else if var_env.contains("uz") {
+        return Language::Uz(region);
+    } else if var_env.contains("vi") {
+        return Language::Vi(region);
+    } else if var_env.contains("vo") {
+        return Language::Vo(region);
+    } else if var_env.contains("cy") {
+        return Language::Cy(region);
+    } else if var_env.contains("wo") {
+        return Language::Wo(region);
+    } else if var_env.contains("xh") {
+        return Language::Xh(region);
+    } else if var_env.contains("ji") {
+        return Language::Ji(region);
+    } else if var_env.contains("yo") {
+        return Language::Yo(region);
+    } else if var_env.contains("zu") {
```

Comment:
As for these methods, I think this needs to be rethought a little.

Once the `LANG SEPARATOR COUNTRY` is returned by the OS, the string should be split, so that there are two variables; `lang` and `country`.  Rather than calling `.contains()`, they should be able to be matched exactly:

```rust
match lang {
    lang if ["AF", "AFG", "004"].contains(&lang) => Language::Af(region),
    _ => /* ... */,
}
```